src/session-pool.ts

```diff
@@ -1036,6 +1151,71 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     session.txn = transaction;
   }
 
+  /**
+   * recycles sessions with long-running transactions
+   *
+   * @private
+   *
+   * @param {number} longRunningTransactionTimeout Timeout for recycling sessions.
+   */
+  async _deleteLongRunningTransactions(
+    longRunningTransactionTimeout: number = 1000 * 60 * 60
+  ): Promise<void> {
+    // Check lock, only one cleanup should take place at a time
+    if (this._ongoingTransactionDeletion) return;
+    this._ongoingTransactionDeletion = true;
+
+    // Stop cleanup after 60 minutes until triggered again unless a session has
+    // been recycled in the last 60 minutes.
+    if (Date.now() - this._lastSessionRecycle! > 60 * 60 * 1000) {
+      this._stopCleaningLongRunningSessions();
+      this._lastSessionRecycle = null;
+      this._ongoingTransactionDeletion = false;
+      return;
+    }
+
+    // Traverse each ongoing transaction and check if it needs to be recycled
+    for (const session of this._traces.keys()) {
+      if (session.lastUsed && !session.longRunningTransaction) {
+        const diff = Date.now() - session?.lastUsed;
+        if (
+          !session.longRunningTransaction &&
+          diff > longRunningTransactionTimeout
+        ) {
+          let sessionTrace = SessionPool.formatTrace(
+            this._traces.get(session) || [],
+            'Long running transaction! Transaction has been closed as it was running for ' +
+              'more than 60 minutes. For long running transactions use batch or partitioned transactions.'
+          );
+          if (this.options.closeInactiveTransactions) {
+            // Recycle session if closeInactiveTransactions is true and log if logging is true.
+            if (this.options.logging) {
+              this.database.logger?.log('warn', sessionTrace);
+            }
+
+            this._recycledTransactions.set(session.txn!, sessionTrace);
```

Comment:
Why do we need to store sessionTrace object with transaction object ?

Reply:
Removing _recycledTransactions